App.js

```diff
@@ -52,10 +53,62 @@ if (
   UIManager.setLayoutAnimationEnabledExperimental(true);
 }
 
+function myiOSPromptCallback(permission) {
+  // do something with permission value
+}
+
+function onReceived(notification) {
+  console.log('Notification received: ', notification);
+}
+
+function onOpened(openResult) {
+  console.log('Message: ', openResult.notification.payload.body);
+  console.log('Data: ', openResult.notification.payload.additionalData);
+  console.log('isActive: ', openResult.notification.isAppInFocus);
+  console.log('openResult: ', openResult);
+}
+
+function onIds(device) {
+  console.log('Device info: ', device);
+}
+
 function App() {
   const [isLoadingComplete, setIsLoadingComplete] = useState(false);
   const [showStorybook, setShowStorybook] = useState(false);
 
+  // Remove this method to stop OneSignal Debugging
+  OneSignal.setLogLevel(6, 0);
+
+  // Replace 'YOUR_ONESIGNAL_APP_ID' with your OneSignal App ID.
+  OneSignal.init(keys.ONESIGNAL_APP_ID, {
+    kOSSettingsKeyAutoPrompt: false,
+    kOSSettingsKeyInAppLaunchURL: false,
+    kOSSettingsKeyInFocusDisplayOption: 2,
+  });
+  OneSignal.inFocusDisplaying(2); // Controls what should happen if a notification is received while the app is open. 2 means that the notification will go directly to the device's notification center.
+
+  // The promptForPushNotifications function code will show the iOS push notification prompt. We recommend removing the following code and instead using an In-App Message to prompt for notification permission (See step below)
+  OneSignal.promptForPushNotificationsWithUserResponse(myiOSPromptCallback);
```

Comment:
are we doin anything with the callback?

Reply:
We are not! At let I'm not sure we need to be. Do we need to amplitude log that?